### ISSUE TITLE
have a question 

### ISSUE THREAD

**Issue by XAB745220zt** [CVLAB-Unibo/Real-time-self-adaptive-deep-stereo#7]
Hello, now I want to use your model to run the video captured by the camera, but I don't know how to do it, can you help me

**Comment by AlessioTonioni** [comment]
You can use placeholders to feed the network  with images acquired by arbitrary source.
A minimal script will look something like this:

```python
left_img_batch = tf.placeholder(tf.float32,shape=[1,384,1280,3], name='left_input')
right_img_batch = tf.placeholder(tf.float32, shape=[1,384,1280,3], name='right_input')

with tf.variable_scope('model'):
    net_args = {}
    net_args['left_img'] = left_img_batch
    net_args['right_img'] = right_img_batch
    net_args['split_layers'] = [None]
    net_args['sequence'] = True
    net_args['train_portion'] = 'BEGIN'
    net_args['bulkhead'] = False
    stereo_net = Nets.get_stereo_net('MADNet', net_args)
    prediction = stereo_net.get_disparities()[-1]
```

Then you can feed your stereo images to the network using `left_img_batch` and `right_img_batch`.

```python
sess = tf.Session()
...
left_image, right_image = ... # read image frome the camera feed
fd = {
     left_img_batch: left_image,
     right_image_batch: right_image
}
disparity = sess.run(prediction,feed_dict=fd)
```

**Comment by XAB745220zt** [comment]
Thank you for your quick reply. I feel very happy, but I have another question

inputs={
			'left':left_img_batch,
			'right':right_img_batch,
			'target':gt_image_batch
		}
If I don't have input   'target':gt_image_batch   Will he make mistakes?

**Comment by AlessioTonioni** [comment]
it is not used for the unsupervised online adaptation so you can pass an empty tensor or something like:

```python
'target':tf.zeros_like(left_img_batch)
```

**Comment by XAB745220zt** [comment]
Thank you, what is this return, because I want to achieve the real world coordinates of the measured object, but the level is limited, I am not understanding what is the depth estimate, what is the stereo match, how to use after getting their output

**Comment by XAB745220zt** [comment]
I really want to get an ideal effect, because I tried to achieve it on opencv, but the effect is very poor, so I turned to deep learning, I don't know how this network works, I also saw more of your excellent projects, such as unsupervised Although our measurement distance is between 0.5m and 3m, it has been bothering me for a long time.
I really need your help.

**Comment by mattpoggi** [comment]
Hi @XAB745220zt ,
if you are wondering how to obtain 3D world coordinates, you can get distance Z from the camera as b*f/d. Then, you can triangulate 2D pixel coordinates to X,Y world coordinates using f and Z.

**Comment by XAB745220zt** [comment]
I am trying to do this, that is, the result of this network is parallax d? Don't need any unit conversions? I don't know how your project Unsupervised-Adaptation-for-Deep-Stereo? Because I don't have a data set for my task to train, I want to try unsupervised or self-monitoring.

**Comment by XAB745220zt** [comment]
Hello, thank you for your patience. When I input the frames acquired by the camera into the input layer of the network, do I need to correct them with the camera's calibration parameters?

**Comment by XAB745220zt** [comment]
Hello, the image I entered is RGB, and the image size is 640X480 and the input is shape=[1,384,1280,3]
I got a night and didn't know how to convert, whether the image needs to be converted to BGR

**Comment by XAB745220zt** [comment]
@mattpoggi @AlessioTonioni 

Hello
Is the parallax d obtained here? Does it need to be converted?

disparity = sess.run(full_res_disp,feed_dict=fd)

**Comment by XAB745220zt** [comment]
@mattpoggi Hello, I changed very badly. Is it because I didn't train? I only loaded your pre-training weights. Another point is that I don't have my own data set to train. My test image comes from the video captured by the camera.

**Comment by AlessioTonioni** [comment]
> Is the parallax d obtained here? Does it need to be converted?

The output of the network is a disparity map aligned with the left input frame. To get per-pixel depth values you can use standard stereo triangulation, [additional info here](https://en.wikipedia.org/wiki/Computer_stereo_vision)

> Hello, thank you for your patience. When I input the frames acquired by the camera into the input layer of the network, do I need to correct them with the camera's calibration parameters?

The left and right frames need to be rectified.

> Hello, the image I entered is RGB, and the image size is 640X480 and the input is shape=[1,384,1280,3]

You should change the input resolution to the one of your camera, i.e. [1,480,640,3]. The channel order is RGB.

> only loaded your pre-training weights. Another point is that I don't have my own data set to train.  My test image comes from the video captured by the camera.

Without any kind of adaptation to the target environment, the performance will be very poor.
You can try to online adapt the network without supervision modifying the code sample in Stereo_Online_Adaètation.py to take the stream from your camera as input.

**Comment by XAB745220zt** [comment]
You told me before:
it is not used for the unsupervised online adaptation so you can pass an empty tensor or something like:

' target '：tf.zeros_like（left_img_batch）

Then tell me:
Without any kind of adaptation to the target environment, the performance will be very poor.
You can try to online adapt the network without supervision modifying the code sample in Stereo_Online_Adaètation.py to take the stream from your camera as input.

I don't quite understand that I don't have training data for my task, only one camera.

**Comment by XAB745220zt** [comment]
@AlessioTonioni  Thank you, choosing the wrong profession is a horrible thing.

**Comment by AlessioTonioni** [comment]
> I don't quite understand that I don't have training data for my task, only one camera.

It should be enough, have you read the paper describing our work? (the link is available in the readme)

We do not use ground truth annotations to adapt the network to the target environment, but compute instead a photoconsistency loss between left input frame and a synthethic left frame obtained by bilinear sampling according to the predicted disparity values.
This loss computation is implemented in `Stereo_Online_Adaptation.py` here:
```python
#compute reprojection error
with tf.variable_scope('reprojection_'+str(counter)):
	reconstruction_loss = loss_factory.get_reprojection_loss('mean_SSIM_l1',reduced=True)([p],inputs_modules)
``` 
The resulting loss do not use 'target' values.

**Comment by XAB745220zt** [comment]
@mattpoggi 

Thank you, I will try it and report it to you.

**Comment by XAB745220zt** [comment]
![disparity_30](https://user-images.githubusercontent.com/35139578/55799926-13ffb500-5b05-11e9-8e2f-9fb614300401.png)
@mattpoggi Thank you, I am getting such a picture, but I have not corrected the picture yet. I still don't understand you very much. I will not update the weight, which means the effect will not be too good, right? Since I don't speak English, I don't think you understand too much about your paper. I am really embarrassed.
